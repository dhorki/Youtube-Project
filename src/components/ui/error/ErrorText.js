import React from 'react';
import PropTypes from 'prop-types';
import { lazyRemoveHtmlTags } from '../../../helpers/decodeHtml';

export const ErrorText = ({ error }) => {
  const errorMessage = lazyRemoveHtmlTags(error?.message);

  return <p className="animate__animated animate__fadeIn">Error: {errorMessage}</p>;
};

ErrorText.propTypes = {
  error: PropTypes.object.isRequired,
};
