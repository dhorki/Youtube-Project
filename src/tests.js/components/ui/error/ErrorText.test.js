import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { ErrorText } from '../../../../components/ui/error/ErrorText';

describe('ErrorText', () => {
  it('should render correctly', () => {
    const error = {
      message: 'Error message',
    };

    const wrapper = shallow(<ErrorText error={error} />);
    expect(wrapper).toMatchSnapshot();
  });
});
