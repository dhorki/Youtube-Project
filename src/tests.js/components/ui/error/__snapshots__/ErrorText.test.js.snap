// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorText should render correctly 1`] = `
<p
  className="animate__animated animate__fadeIn"
>
  Error: 
  Error message
</p>
`;
